blurring_as_a_service/metrics/custom_metrics_calculator.py

```diff
@@ -0,0 +1,500 @@
+                "| False Negatives | FN Small | FN Medium | FN Large | False Negative Rate | False Negative Rate Small "
+                "|False Negative Rate Medium | False Negative Rate Large|\n"
+            )
+            f.write(
+                "| -------- | ----- | ----- | ----- | ----- | ----- | ----- | ----- | ----- | ----- | ----- |"
+                "-------------- | --------------- | ------------------ |\n"
+            )
+            for item in result:
+                f.write(
+                    f'| {item["category"]} | {item["value"]} | {item["true_positives"]} '
+                    f'| {item["true_positives_small"]} | {item["true_positives_medium"]} | {item["true_positives_large"]} '
+                    f'| {item["false_negatives"]} | {item["false_negatives_small"]} | {item["false_negatives_medium"]} '
+                    f'| {item["false_negatives_large"]} | {item["false_negative_rate"]}'
+                    f'| {item["false_negative_rate_small"]} | {item["false_negative_rate_medium"]}'
+                    f'| {item["false_negative_rate_large"]}\n'
+                )
+            f.write(
+                f"Thresholds used for these calculations: Small=`{Size.small.value}`, Medium=`{Size.medium.value}` "
+                f"and Large=`{Size.large.value}`."
+            )
+
+    def calculate_false_negative_rate_for_all_categories(self) -> List[dict]:
+        """
+        Using the calculate_false_negative_rate_for_a_category function, it calculates the value for the following categories:
+            - 'grouped_category': Persons or licence plates.
+            - 'sex'.
+            - 'age'.
+            - 'skin_color'.
+        Returns
+        -------
+        A list where each element is a category and its metrics, included the false negative rate.
+        """
+        result = []
+        for category in [
+            "grouped_category",
+            "sex",
+            "age",
+            "skin_color",
+            "licence_plate_origin",
+            "licence_plate_color",
+        ]:
+            print(category)
+            result.append(self.calculate_false_negative_rate_for_a_category(category))
+        return list(itertools.chain(*result))
+
+    def calculate_false_negative_rate_for_a_category(
+        self, category_name: str
+    ) -> List[dict]:
+        """
+        Given a category name it calculates the false negative rate of that category
+        checking the values in _statistics_per_category.
+
+        Parameters
+        ----------
+        category_name
+
+        Returns
+        -------
+        A list where each element is a category value and its metrics, included the false negative rate.
+        """
+        available_categories = list(
+            {v[category_name] for k, v in self._statistics_per_category.items()}
+        )
+        all_categories = {
+            available_category: {
+                "true_positives": 0,
```

Comment:
Maybe also for this? And finding a better name, I have the feeling everything is `category` at the moment.